eclair-core/src/main/scala/fr/acinq/eclair/channel/Helpers.scala

```diff
@@ -189,25 +188,20 @@ object Helpers {
   }
 
   /**
-   * @param referenceFeePerKw reference fee rate per kiloweight
-   * @param currentFeePerKw   current fee rate per kiloweight
-   * @return the "normalized" difference between i.e local and remote fee rate: |reference - current| / avg(current, reference)
+   * To avoid spamming our peers with fee updates every time there's a small variation, we only update the fee when the
+   * difference exceeds a given ratio (updateFeeMinDiffRatio).
    */
-  def feeRateMismatch(referenceFeePerKw: Long, currentFeePerKw: Long): Double =
-    Math.abs((2.0 * (referenceFeePerKw - currentFeePerKw)) / (currentFeePerKw + referenceFeePerKw))
-
-  def shouldUpdateFee(commitmentFeeratePerKw: Long, networkFeeratePerKw: Long, updateFeeMinDiffRatio: Double): Boolean =
-    feeRateMismatch(networkFeeratePerKw, commitmentFeeratePerKw) > updateFeeMinDiffRatio
+  def shouldUpdateFee(currentFeeratePerKw: Long, nextFeeratePerKw: Long, updateFeeMinDiffRatio: Double): Boolean =
+    Math.abs((currentFeeratePerKw - nextFeeratePerKw).toDouble / currentFeeratePerKw) > updateFeeMinDiffRatio
```

Comment:
We should check the division by zero, it isn't much effort.

Reply:
Fixed in https://github.com/ACINQ/eclair/pull/1473/commits/9037da2978266fb0de40bf13c272d0024dfedcea